File: Ugly Number II.cpp
class Solution {
public:
	int nthUglyNumber(int n) {
		vector<int> res ;

		res.push_back(1) ;

		int i = 0 , j = 0 , k = 0 , t = n ;

		while (n--) {

			int min1 = 2 * res[i] ;
			int min2 = 3 * res[j] ;
			int min3 = 5 * res[k] ;

			int minm = min(min1, min(min2, min3)) ;

			res.emplace_back(minm) ;

			if (minm == min1) i++ ;
			if (minm == min2) j++ ;
			if (minm == min3) k++ ;
		}

		return res[t - 1] ;
	}
};

File: Ugly Number II.java
class Solution {
    public int nthUglyNumber(int n) {
        ArrayList<Integer> myvec = new ArrayList<Integer>();

        Integer i = 0 , j = 0 , k = 0 ;
        Integer t = n + 2 ;
        myvec.add(1) ;

        while (myvec.size() < t) {
            Integer min2 = myvec.get(i) * 2 ;
            Integer min3 = myvec.get(j) * 3;
            Integer min5 = myvec.get(k) * 5;

            Integer mynum = Math.min(min2, Math.min(min3, min5)) ;

            myvec.add(mynum) ;
            if (mynum == min2)
                i++ ;
            if (mynum == min3)
                j++ ;
            if (mynum == min5)
                k++ ;
        }

        return myvec.get(n - 1) ;
    }
}